x-pack/plugin/search-business-rules/src/main/java/org/elasticsearch/xpack/searchbusinessrules/PinnedQueryBuilder.java

```diff
@@ -105,13 +102,13 @@ private Item(String id) {
          * Read from a stream.
          */
         public Item(StreamInput in) throws IOException {
-            index = in.readString();
+            index = in.readOptionalString();
             id = in.readString();
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeString(index);
```

Comment:
This changes serialization. I guess we haven't released this yet, so maybe this doesn't apply - but shouldn't we bump `TransportVersion` and use `readString()` or `readOptionalString()` depending on its value? 🤔

Reply:
💯 Yes, we need to do this.

Reply:
@kderusso , what's the status for this PR? Should we work on this to push it through the finish line?